robottelo/rerun_failures/uncollector.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+import pytest
+
+from robottelo.config import settings
+from robottelo.report_portal.portal import ReportPortal
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_launch(rp):
+    """Returns the latest launch of satellite"""
+    _version = settings.server.version
+    sat_version = f'{_version.base_version}.{_version.epoch}'
+    launch = next(iter(rp.launches(sat_version=sat_version).values()))
+    # Stop session based on RP Launch statistics and info
+    pass_percent = round(
+        100 * (int(launch.statistics['passed']) / int(launch.statistics['total']))
+    )
+    if launch.info['isProcessing']:
+        pytest.skip(f'The latest launch of satellite version {sat_version} is not Finished yet.')
+    if pass_percent < 80:
+        pytest.skip(
+            f'The latest Satellite launch has only {pass_percent}% tests passed. '
+            'Which is lesser than the threshold of 80%. '
+            'Examine the failures thoroughly and check for any major issue.'
+        )
+    return launch
+
+
+def _get_failed_tests(launch, fail_args):
+    """Returns failed test names from Report Portal Launch based on arguments"""
+    tests = []
+    if fail_args == ['all']:
+        test_names = [*launch.tests(status='failed').keys()]
+        tests.extend(test_names)
+    else:
+        for dtype in fail_args:
+            test_names = [*launch.tests(defect_type=dtype).keys()]
+            tests.extend(test_names)
```

Comment:
Consider structuring the logic a bit differently here to highlight what is actually different, since the same actions are being taken.

```
# typing this without running it, so use as example not exact code

test_args = [dict(status='failed')] if fail_args.pop() == 'all' else [dict(defect_type=dtype) for dtype in fail_args]
tests = [*launch.tests(**t_args).keys() for t_args in test_args]

# then do string replace and return
```

Reply:
The *ing outside the inner list didnt work. The inner_list * works but the issue here the square brackets of inner list are preserved during list comprehension and its creating list of list. Hence went back to the existing method of extending.